### PR TITLE
[ETFE-3426] alignments to CAM-ITM42 - clear input, inset, hint, etc

### PR DESCRIPTION
**Pull request by james-work-account**

### CODE REVIEW

File: app/models/requests/DataRequest.scala [hunk 1]
```diff
@@ -50,4 +52,19 @@ case class DataRequest[A](request: UserRequest[A],
...
+    (userAnswers.get(DestinationTypePage), userAnswers.get(GuarantorRequiredPage)) match {
+      case (Some(scenario), Some(false)) if scenario.movementType(this) == MovementType.UkToUk => true
+      case _ => false
+    }
+
+  def isUkToEuAndNoGuarantor: Boolean =
+    (userAnswers.get(DestinationTypePage), userAnswers.get(GuarantorRequiredPage)) match {
+      case (Some(scenario), Some(false)) if scenario.movementType(this) == MovementType.UkToEu => true
+      case _ => false
+    }
```
Comment (MJCallahanPage) [inline app/models/requests/DataRequest.scala:66]: I'm not sure this check is right, shouldn't this be only for NI to EU? Also, there are other movement types which would count as NI to EU which aren't covered by the `MovementType.UKtoEU`

**Review by MJCallahanPage — CHANGES_REQUESTED** [review]
I know you've copied what's already there, but I think there might be a bug that the `ukToEUNoGuarantor` check should actually only be for `NItoEU` - maybe need to check with Lexi/Haresh.

File: app/models/requests/DataRequest.scala (hunk 1 above)
Comment (james-work-account) [reply app/models/requests/DataRequest.scala:66]: Yeah I copied the existing code, I'll double check if it's actually right though with Lexi/Haresh.

File: app/models/UserAnswers.scala [hunk 2]
```diff
@@ -79,6 +79,16 @@ final case class UserAnswers(ern: String,
       data.setObject(section.path, Json.obj())
     }
 
+  /**
+   * @param section section to reset to an empty object
+   * @return UserAnswers with the supplied section reset to an empty JSON object
+   *         Note, if used on an indexed section, the index will be lost - use resetIndexedSection for indexed sections
```
Comment (james-work-account) [inline app/models/UserAnswers.scala:85]: I'm aware this comment is wrong, need to go back and change it at some point

**Review by MJCallahanPage — APPROVED** [review]
LGTM 🟢

File: app/models/UserAnswers.scala [hunk 3]
```diff
@@ -79,6 +79,16 @@ final case class UserAnswers(ern: String,
...
+   */
+  def resetSection(section: Section[JsObject]): UserAnswers =
```
[the remark this answers is not in the document]
Comment (MJCallahanPage) [reply app/models/UserAnswers.scala]: ignore this, you already delete it.

**Pull request merged**